src/backlight/backlight.c

```diff
@@ -379,7 +379,7 @@ static int run(int argc, char *argv[]) {
                 clamp = shall_clamp(device);
 
                 r = read_one_line_file(saved, &value);
-                if (r == -ENOENT) {
+                if (r == -ENOENT || r == 0) {
```

Comment:
`read_one_line_file()` returns `0` on success, not a number of read characters. So, this PR always override `value` by the fallback logic below...

Reply:
Sorry, I looked at wrong function.

It seems that `read_one_line_file()` could be changed easily to return number of bytes read on success. All users of it already check for < 0 for failure and >= 0 for success, with the exception of basic/virt.c:202.

Alternatively there could be explicit size checks with `stat()` or so but I think this would be simpler.

Reply:
Here are some other users of `read_one_line_file()` which could also benefit if they could reject zero sized reads early:

https://github.com/systemd/systemd/blob/118dccc9489829ad74a5d1063ee7fe7e3b86cf62/src/basic/virt.c#L471
https://github.com/systemd/systemd/blob/118dccc9489829ad74a5d1063ee7fe7e3b86cf62/src/core/dynamic-user.c#L709
https://github.com/systemd/systemd/blob/118dccc9489829ad74a5d1063ee7fe7e3b86cf62/src/cryptsetup/cryptsetup.c#L450
https://github.com/systemd/systemd/blob/118dccc9489829ad74a5d1063ee7fe7e3b86cf62/src/firstboot/firstboot.c#L862
https://github.com/systemd/systemd/blob/118dccc9489829ad74a5d1063ee7fe7e3b86cf62/src/rfkill/rfkill.c#L149
https://github.com/systemd/systemd/blob/118dccc9489829ad74a5d1063ee7fe7e3b86cf62/src/shared/dissect-image.c#L1273
https://github.com/systemd/systemd/blob/118dccc9489829ad74a5d1063ee7fe7e3b86cf62/src/shared/reboot-util.c#L52

I assume APIVFS files are never truncated.

Reply:
> It seems that `read_one_line_file()` could be changed easily to return number of bytes read on success.

I like this direction, though I have not checked the list you've shown in the above.